hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/ReplicatedMapService.java

```diff
@@ -16,86 +16,152 @@
+    private final ReplicatedMapEventPublishingService eventPublishingService;
+    private final MergePolicyProvider mergePolicyProvider;
+    private final ReplicatedMapSplitBrainHandlerService replicatedMapSplitBrainHandlerService;
+    private ConcurrentHashMap<String, LocalReplicatedMapStatsImpl> statsMap =
+            new ConcurrentHashMap<String, LocalReplicatedMapStatsImpl>();
+    private ConstructorFunction<String, LocalReplicatedMapStatsImpl> constructorFunction =
+            new ConstructorFunction<String, LocalReplicatedMapStatsImpl>() {
+                @Override
+                public LocalReplicatedMapStatsImpl createNew(String arg) {
+                    return new LocalReplicatedMapStatsImpl();
+                }
+            };
 
     public ReplicatedMapService(NodeEngine nodeEngine) {
         this.nodeEngine = nodeEngine;
         this.config = nodeEngine.getConfig();
-        this.eventService = nodeEngine.getEventService();
-        if (!config.isLiteMember()) {
-            this.eventRegistration = eventService.registerListener(SERVICE_NAME, EVENT_TOPIC_NAME, new ReplicationListener());
-        } else {
-            this.eventRegistration = null;
-        }
+        this.partitionService = (InternalPartitionServiceImpl) nodeEngine.getPartitionService();
+        this.clusterService = (ClusterServiceImpl) nodeEngine.getClusterService();
+        this.operationService = nodeEngine.getOperationService();
+        this.partitionContainers = new PartitionContainer[nodeEngine.getPartitionService().getPartitionCount()];
+        this.eventPublishingService = new ReplicatedMapEventPublishingService(this);
+        this.mergePolicyProvider = new MergePolicyProvider(nodeEngine);
+        this.replicatedMapSplitBrainHandlerService = new ReplicatedMapSplitBrainHandlerService(this,
+                mergePolicyProvider, partitionContainers);
     }
 
     @Override
-    public void init(NodeEngine nodeEngine, Properties properties) {
+    public void init(final NodeEngine nodeEngine, Properties properties) {
+        if (config.isLiteMember()) {
+            return;
+        }
+
+        for (int i = 0; i < nodeEngine.getPartitionService().getPartitionCount(); i++) {
+            partitionContainers[i] = new PartitionContainer(this, i);
+        }
+        nodeEngine.getExecutionService().getDefaultScheduledExecutor().scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                if (clusterService.getSize() == 1) {
+                    return;
+                }
+                Collection<Address> addresses = new ArrayList<Address>(getMemberAddresses(DATA_MEMBER_SELECTOR));
+                addresses.remove(nodeEngine.getThisAddress());
+                for (int i = 0; i < partitionContainers.length; i++) {
+                    Address thisAddress = nodeEngine.getThisAddress();
+                    Address ownerAddress = partitionService.getPartitionOwner(i);
```

Comment:
This line triggers the partition assignment even if there's no data. You can change it to:

```java
InternalPartition partition = partitionService.getPartition(i, false);
Address ownerAddress = partition.getOwnerOrNull();
```